datafusion/common/src/utils/mod.rs

```diff
@@ -741,6 +741,43 @@ pub mod datafusion_strsim {
         generic_levenshtein(&StringWrapper(a), &StringWrapper(b))
     }
 
+    /// Calculates the Levenshtein distance using a reusable cache buffer.
+    /// This avoids allocating a new Vec for each call, improving performance
+    /// when computing many distances.
+    ///
+    /// The `cache` buffer will be resized as needed and reused across calls.
+    pub fn levenshtein_with_buffer(a: &str, b: &str, cache: &mut Vec<usize>) -> usize {
```

Comment:
I wonder if there's a way to unify this with `generic_levenshtein` to avoid duplication, something like having `generic_levenshtein` call this but creating a vec buffer at the callsite? 🤔